### DISCUSSION TITLE
How to get Options for specific date in options module

### DISCUSSION THREAD

**Discussion by trungisme** [gadicc/yahoo-finance2#286] [Q&A] [answered]
Hi! This is a great module so far, I've been reading through it and have found the features so far quite extensive and useful!

I am wondering how to get options data for specific date but not as a quote. 
Based on the request that is being sent I want to get the monthly expiration option chains which can be be received by hitting the request: https://query1.finance.yahoo.com/v7/finance/options/AAPL?date=1631836800 
![image](https://user-images.githubusercontent.com/14306946/132568140-5388160f-2e92-4f11-a167-e93e2b79b71e.png)

By default the service gets the nearest expiration options chain:
![image](https://user-images.githubusercontent.com/14306946/132568243-634d06e7-13b0-4e38-a76d-5bfe2f0c96ec.png)

Based on the codebox example available at: https://codesandbox.io/s/yahoo-finance2-forked-5k9kn?file=/src/index.ts, I have tried the following:
```
import yahooFinance from "yahoo-finance2";
import { show } from "./utils";

show(async () => {
  const queryOptions = { lang: "en-US", formatted: false, region: "US" };
  const result = await yahooFinance.options("AAPL", queryOptions, {
    validateResult: false,
    devel: false,
    fetchOptions: {
      date: "1631836800"
    }
  });
  return result;

});
```

I am struggling to find a way to do this as I am new to typescript and can't read the information quite well. Any help would be greatly appreciated!

**Comment by trungisme** [comment, marked as answer]
Hi!
I managed to resolve this issue by updating the queryOptions instead of the moduleOptions:
```
import yahooFinance from "yahoo-finance2";
import { show } from "./utils";

show(async () => {
  const queryOptions = { lang: "en-US", formatted: false, region: "US" };
  const result = await yahooFinance.options("AAPL", {
    date: 1631836800
  });
  return result;
});
```

I found this from here:
https://github.com/gadicc/node-yahoo-finance2/blob/c01fe74bcbab289ccf693ffe4887303a529bd584/schema.json#L3420-L3437

**Reply by gadicc** [reply]
Hey, @trungisme. Great job on answering your own question! These are always our favourite kinds of discussions :)

So thanks for all the research you did and figuring this out.

I will note that it seems we made a small mistake here, in that the date option should allow for a date type and not a number. I opened #287 for that, also with a note to add this to the docs to make it easier to find for others.

Thanks!

**Reply by gadicc** [reply]
Note also in your code above, you don't need the `const queryOptions = ` line since you're not using it and have retained all the defaults.  Good luck!
